dist/cargo_test.sh

```diff
@@ -23,16 +23,29 @@ executors["cargo"]="execute_native"
 executors["docker"]="execute_docker"
 
 function run_tests() {
+  HAS_KOV=0
+  if [[ $(type -f kcov && cargo kcov --version) ]]; then HAS_KOV=1; fi
   for directory in ${!cargo_test_flags[*]}; do
     (
-      set -x;
+      set -x
+      if [[ "${HAS_KOV}" ]]; then
+        # we want to prevent completely untested function to be stripped
+        export RUSTFLAGS='-C link-dead-code'
+        export CARGO_ARGS="kcov --verbose --all --no-clean-rebuild"
+      fi
+
       ${1} /usr/bin/env bash -c "\
         cd ${directory} && \
         export CARGO_TARGET_DIR=\"../target\" && \
+        cargo test --release --no-run &&
```

Comment:
This compilation of the tests in `--release` mode is superfluous. The next line will compile and run the tests accordingly.

```suggestion
```

Reply:
Turns out `cargo kcov` needs it - it won't build tests itself